packages/trader/src/App/Components/Form/CompositeCalendar/composite-calendar-mobile.jsx

```diff
@@ -112,7 +112,23 @@ const CompositeCalendarMobile = React.memo(
             }
 
             if (key === 'to') {
-                setTo(e.target?.value ? toMoment(e.target.value).format('DD MMM YYYY') : '');
+                // Handling future dates input for IOS devices
+                const value = e.target?.value ? toMoment(e.target.value).format('DD MMM YYYY') : '';
+
+                setTo(value);
+
+                if (value) {
+                    const upcoming_days = parseInt(daysFromTodayTo(value));
+                    if (!isNaN(upcoming_days)) {
+                        setTo('');
+
+                        const today = toMoment().format('DD MMM YYYY');
+
+                        setTimeout(() => {
+                            setTo(today);
```

Comment:
What is the reason to set a timeout here? Is that because of `toMoment` response delay?

Reply:
The delay creates an effect to force the selected future dates back to current date. without the delay on IOS devices it will only force the future dates to current date only once.